src/components/Trade/TradeInput.tsx

```diff
@@ -47,28 +54,125 @@ export const TradeInput = ({ history }: RouterProps) => {
     number: { localeParts, toFiat },
   } = useLocaleFormatter()
   const [isSendMaxLoading, setIsSendMaxLoading] = useState<boolean>(false)
-  const [quote, buyTradeAsset, sellTradeAsset, feeAssetFiatRate, quoteError] = useWatch({
-    name: ['quote', 'buyAsset', 'sellAsset', 'feeAssetFiatRate', 'quoteError'],
-  }) as [TS['quote'], TS['buyAsset'], TS['sellAsset'], TS['feeAssetFiatRate'], TS['quoteError']]
-  const {
-    updateQuote,
-    checkApprovalNeeded,
-    getSendMaxAmount,
-    updateTrade,
-    feeAsset,
-    refreshQuote,
-  } = useSwapper()
+  const [
+    quote,
+    buyTradeAsset,
+    sellTradeAsset,
+    feeAssetFiatRate,
+    quoteError,
+    sellAssetAccount,
+    selectedAssetAccount,
+    action,
+    amount,
+  ] = useWatch({
+    name: [
+      'quote',
+      'buyAsset',
+      'sellAsset',
+      'feeAssetFiatRate',
+      'quoteError',
+      'sellAssetAccount',
+      'selectedAssetAccount',
+      'action',
+      'amount',
+    ],
+  }) as [
+    TS['quote'],
+    TS['buyAsset'],
+    TS['sellAsset'],
+    TS['feeAssetFiatRate'],
+    TS['quoteError'],
+    TS['sellAssetAccount'],
+    TS['selectedAssetAccount'],
+    TS['action'],
+    TS['amount'],
+  ]
+  const { updateQuote, checkApprovalNeeded, getSendMaxAmount, updateTrade, feeAsset } = useSwapper()
   const translate = useTranslate()
   const selectedCurrencyToUsdRate = useAppSelector(selectFiatToUsdRate)
   const {
     state: { wallet },
   } = useWallet()
 
-  const sellAssetBalance = useAppSelector(state =>
-    selectPortfolioCryptoHumanBalanceByAssetId(state, {
-      assetId: sellTradeAsset?.asset?.assetId ?? '',
+  const accountIds = useAppSelector(state => {
+    const assetId = sellTradeAsset?.asset?.assetId
+    return assetId ? selectAccountIdsByAssetId(state, { assetId }) : []
+  })
```

Comment:
As discussed, we shouldn't put such logic inside a selector, since they recompute on inputs change - `sellTradeAsset?.asset?.assetId` not being an input and living inside `useAppSelector()` callback would not recompute the selector - thus not giving a new `accountIds` output and reference - the way we want.

Similarly, and following the same idea of `useAppSelector` callbacks consuming a selector and nothing else, either consuming a selector or returning an empty array might produce unpredictable results. Looking at the implementation of  `selectAccountIdsByAssetId` and its underlying selectors, we should be able to pass `assetId: assetId ?? ''` - that will still return an empty array, although a reference to the right one.

Let's make this line live outside the selector

```suggestion
  const assetId = sellTradeAsset?.asset?.assetId
  const accountIds = useAppSelector(state => selectAccountIdsByAssetId(state, { assetId: assetId ?? '' })
  )
```

Reply:
Appreciate the explanation sir, makes total sense! 

Addressed in https://github.com/shapeshift/web/pull/2471/commits/f91e53de5719375b81d56ddf8d2aa6520c0e67cd

Reply:
Note, I had to add a truthy check on `assetId` to `findAccountsByAssetId` in https://github.com/shapeshift/web/pull/2471/commits/0b30db138f0370d167f3a930a607b742d839fa84 to stop it from throwing when an empty string was provided.